compiler/rustc_passes/src/dead.rs

```diff
@@ -793,6 +793,15 @@ fn check_item<'tcx>(
             // global_asm! is always live.
             worklist.push((id.owner_id.def_id, ComesFromAllowExpect::No));
         }
+        DefKind::Const => {
+            let item = tcx.hir_item(id);
+            if let rustc_hir::ItemKind::Const(ident, _, _, _) = item.kind
+                && ident.name == kw::Underscore
+            {
+                // `const _` is always live.
```

Comment:
Suggestion: could you add a little more context on the *why*? I.e. some TL;DR of https://github.com/rust-lang/rust/issues/142075#issuecomment-2944261976.

Reply:
Sure, done.